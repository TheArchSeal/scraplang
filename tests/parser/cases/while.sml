while (x);
if (x) while (x) if (x); else; else;
do x; while (x);
do while (x) x; while (x);
