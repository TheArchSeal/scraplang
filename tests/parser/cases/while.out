stmt (8):1:1 while
    expr (3):1:8 x
    stmt (1):1:10 (nop)
stmt (6):2:1 if else
    expr (3):2:5 x
    stmt (8):2:8 while
        expr (3):2:15 x
        stmt (6):2:18 if else
            expr (3):2:22 x
            stmt (1):2:24 (nop)
            stmt (1):2:30 (nop)
    stmt (1):2:36 (nop)
stmt (9):3:1 do while
    expr (3):3:14 x
    stmt (3):3:4 ;
        expr (3):3:4 x
stmt (9):4:1 do while
    expr (3):4:24 x
    stmt (8):4:4 while
        expr (3):4:11 x
        stmt (3):4:14 ;
            expr (3):4:14 x
